File: t/PACSFramework/CS/StorageServerManager/DataTypes/Options/PasswordOptions.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Leadtools.Demos.StorageServer.DataTypes.Options
{
   [Flags]
   public enum Complexity
   {
      None = 0,
      Lowercase = 1,
      Uppercase = 2,
      Symbol = 4,
      Numbers = 8
   }

   [Serializable]
   public class PasswordOptions : ICloneable
   {
      public Complexity Complexity { get; set; }
      public int MinimumLength { get; set; }
      public int DaysToExpire { get; set; }
      public int MaxPasswordHistory { get; set; }
      public int IdleTimeOut { get; set; }
      public bool EnableIdleTimeout { get; set; }

      public PasswordOptions()
      {
         IdleTimeOut = 60;
         EnableIdleTimeout = true;
      }

      #region ICloneable Members

      public object Clone()
      {
         return this.Clone<PasswordOptions>();
      }

      #endregion
   }
}

File: t/PACSFramework/VB/StorageServerManager/DataTypes/Options/PasswordOptions.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Leadtools.Demos.StorageServer.DataTypes.Options
   <Flags()> _
   Public Enum Complexity
      None = 0
      Lowercase = 1
      Uppercase = 2
      Symbol = 4
      Numbers = 8
   End Enum

   <Serializable()> _
   Public Class PasswordOptions : Implements ICloneable
      Public Property Complexity() As Complexity
         Get
         Set
         End Set
         End Get
      public Integer MinimumLength
         Get
         Set
         End Set
         End Get
      public Integer DaysToExpire
         Get
         Set
         End Set
         End Get
      public Integer MaxPasswordHistory
         Get
         Set
         End Set
         End Get
      public Integer IdleTimeOut
         Get
         Set
         End Set
         End Get
      public Boolean EnableIdleTimeout
         Get
         Set
         End Set
         End Get

      Public PasswordOptions()
         IdleTimeOut = 60
         EnableIdleTimeout = True

      public Object Clone()
         Return Me.Clone(Of PasswordOptions)()

